microcosm_pubsub/codecs.py

```diff
@@ -99,10 +78,16 @@ def encode(self, dct=None, **kwargs):
         """
         message = dct.copy() if dct else dict()
         message.update(kwargs)
-        # We allow unknown fields to pass through (but not included here)
-        # to accommodate unconditional parameter passing during produce()
-        # e.g. passing in `uri` for IdentityMessages
-        return dumps(self.schema.load(message, unknown=EXCLUDE))
```

Comment:
This was the main crux of the issue I was describing wrt lack of type freedom:
* we were sending the decoded version of the resource over the wire
* this means that custom types that defined a de/serialization method wouldn't even have the chance to run
* this is brain-melting; the encoded form is what's transferred, the decoded form is what's parsed. We shouldn't be confusing the two.